spec/index.md

```diff
@@ -526,7 +526,26 @@ x = true
 
 ### 5.1.1.2. Contract type
 
-An *contract type* is the type denotated by the name of an [`contract` item].
+An *contract type* is the type denoted by the name of an [`contract` item].
+
+A value of a given contract type caries the contract's public interface as 
+attribute functions. A new contract value can be created by either casting
+an address to a contract type or by creating a new contract using the type 
+attribute functions `create` or `create2`.
+
+example:
+
+```python 
+ contract Foo:
+    pub def get_my_num() -> u256:
+        return 42
+
+contract FooFactory:
+    pub def create2_foo() -> address:
+        # `0` is the value being sent and `52` is the address salt
+        foo: Foo = Foo.create2(0, 52)
+        return address(foo)
```

Comment:
Should we consider giving contracts a special attribute to obtain the address such so that the above could be written as:

```
pub def create2_foo() -> address:
        # `0` is the value being sent and `52` is the address salt
        foo: Foo = Foo.create2(0, 52)
        return foo.__address__
```

With the cast, a bug such as the following would be possible that would result in returning the wrong address.

```
pub def create2_foo() -> address:
        # `0` is the value being sent and `52` is the address salt
        foo: Foo = Foo.create2(0, 52)
        bar: Bar = Bar.create2(0, 11)
        return address(bar)
```

In the example above, we *mean* to return the address of the newly created `foo` but we accidentally return the address of `bar` instead. Sure, one can also accidentially return `bar.__address__` but it feels to me that having the address accessible as a property removes one layer of error potential. That said, I really dislike the `__address__` syntax but wouldn't now a good way to get around the potential for naming conflicts. At least not until we have something like traits that we could bring into scope conditionally to expose special functions/attributes.

Reply:
Yeah, I wasn't sure how to deal with this.

We can discuss this more in #179 and if we come up with a better idea, it can be changed. For now, I'm inclined to just merge this as is.

Reply:
Yeah, should be good for now. :+1: